Minigames/src/main/java/au/com/mineauz/minigames/objects/OfflineMinigamePlayer.java

```diff
@@ -36,37 +41,43 @@ public OfflineMinigamePlayer(UUID uuid, ItemStack[] items,
         this.lastGM = lastGM;
         this.exp = exp;
         this.level = level;
-        if (loginLocation != null && loginLocation.getWorld() == null)
-            loginLocation = Bukkit.getWorlds().get(0).getSpawnLocation();
-        this.loginLocation = loginLocation;
+        if (loginLocation != null && loginLocation.getWorld() == null) {
+            this.loginLocation = Bukkit.getWorlds().get(0).getSpawnLocation();
+        } else {
+            this.loginLocation = loginLocation;
+        }
         if (Minigames.getPlugin().getConfig().getBoolean("saveInventory"))
             savePlayerData();
     }
 
-    public OfflineMinigamePlayer(UUID uuid) {
-        MinigameSave save = new MinigameSave("playerdata/inventories/" + uuid.toString());
+    /**
+     * loads player data from disk
+     *
+     * @param uuid the uuid of the user to load
+     */
+    public OfflineMinigamePlayer(final @NotNull UUID uuid) {
+        MinigameSave save = new MinigameSave("playerdata/inventories/" + uuid);
         FileConfiguration con = save.getConfig();
         this.uuid = uuid;
-        food = con.getInt("food");
-        health = con.getDouble("health");
-        saturation = con.getInt("saturation");
+        food = con.getInt("food", 20);
+        health = con.getDouble("health", 20);
+        saturation = con.getInt("saturation", 15);
```

Comment:
Also bad idea to hardcode. Is it possible to get attributes for an offline player?

Reply:
Nope there is no easy way without nms as fas as I can see.
Probably already thought about it, when I made this change.